src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java

```diff
@@ -166,12 +166,12 @@ private JiraSendInfoResponse handleDeploymentApiResponse(
             return JiraDeploymentInfoResponse.successDeploymentAccepted(jiraSite, response);
         }
 
-        if (!response.getRejectedDeployments().isEmpty()) {
-            return JiraDeploymentInfoResponse.failureDeploymentdRejected(jiraSite, response);
+        if (!response.getUnknownAssociations().isEmpty()) {
+            return JiraDeploymentInfoResponse.failureUnknownAssociations(jiraSite, response);
         }
 
-        if (!response.getUnknownIssueKeys().isEmpty()) {
-            return JiraDeploymentInfoResponse.failureUnknownIssueKeys(jiraSite, response);
+        if (!response.getRejectedDeployments().isEmpty()) {
+            return JiraDeploymentInfoResponse.failureDeploymentRejected(jiraSite, response);
```

Comment:
I've switched these checks because of unknows associations was unreachable since an error response always contains rejected deployments. Now Jenkins logs will have this log

```
[Pipeline] jiraSendDeploymentInfo
jiraSendDeploymentInfo: FAILURE_UNKNOWN_ASSOCIATIONS: Failed to send deployment information to Jira: cloud-plugin.atlassian.net. Unknown associations: [Association{associationType=issueKeys, values=[TEST-7]}].
```

instead of

```
[Pipeline] jiraSendDeploymentInfo
jiraSendDeploymentInfo: FAILURE_DEPLOYMENT_REJECTED: Failed to send deployment information to Jira: cloud-plugin.atlassian.net. Rejected deployments: [RejectedDeploymentResponse{key=DeploymentKeyResponse{pipelineId='jenkins-plugin-testing'environmentId='us-tst-1', deploymentSequenceNumber=14}, errors=[ApiErrorResponse{message='No valid associations found for deployment'}]}].
```

Reply:
> an error response always contains rejected deployments
WDYM?

Reply:
Typical error jira deployment response with the wrong associations looks like 

```
{
    "acceptedDeployments": [],
    "rejectedDeployments": [
        {
            "key": {
                "pipelineId": "temp",
                "environmentId": "us-stg-1",
                "deploymentSequenceNumber": 80
            },
            "errors": [
                {
                    "message": "No valid associations found for deployment"
                }
            ]
        }
    ],
    "unknownIssueKeys": [],
    "unknownAssociations": [
        {
            "associationType": "issueKeys",
            "values": [
                "TEST-11"
            ]
        }
    ]
}
```
in case of unknown associations error, a user got a `FAILURE_DEPLOYMENT_REJECTED` error with *No valid associations found for deployment* message. I've changed this behavior to `FAILURE_UNKNOWN_ASSOCIATIONS` *Unknown associations: [Association{associationType=issueKeys, values=[TEST-7]}* for other cases everything works as b4.